upgrade to strong-agent v2

rmg opened the pull request:
Required to get a non-strongops version of strong-agent.

rmg commented:
@slnode test please

rmg commented:
@slnode test please

rmg commented:
@slnode test please

rmg commented:
@slnode test please

sam-github commented on bin/sl-run.js:
```diff
@@ -24,6 +24,9 @@ var agentOptions = {
   logger: config.logger,
   strongTracer: tracer(),
 };
+if (process.env.SPEED_UP_INTERVAL_FOR_TESTS === 'yes') {
```
how about just making a `STRONGLOOP_BASE_INTERVAL` env var, and supporting it at: https://github.com/strongloop/strongops/blob/master/lib/config.js#L5

I think we should STRONGLOOP prefix our env vars, even if they  are sentences, and putting it in lib/config.js makes it easy to find, and useful to agent tests, etc.

rmg replied:
strong-agent already supports an interval control interface, I just consider this a strong-supervisor specific convenience wrapper for it. I don't want to invent another cross-module environment variable to replace the previous one and just repeat this process in 6 months.

sam-github commented:
LGTM, other that I think the env var is in the wrong place.

rmg commented:
My response about the env var got swallowed when I pushed a fixup.

My intention with the env var was to **not** replace an unofficial cross-module interface that broke with another unofficial cross-module interface and instead use the existing interface that is part of the actual API. I extended the name to make it even more obvious what it is for and less likely to be triggered directly.

rmg commented:
changed env var, rebased, squashed..

@sam-github is this what you were thinking of for the interval?

I'm thinking since the interval is no longer something the agent itself actually cares about (because the collector did), the default interval here isn't really a duplicate constant so much as coincidentally identical to the current default since this is the supervisor's default, not the agent's.

rmg commented:
@slnode test please

sam-github commented:
LGTM
